src/h5web/visualizations/shared/hooks.ts

```diff
@@ -1,12 +1,25 @@
 import ndarray from 'ndarray';
-import { useMemo, useState } from 'react';
+import { useMemo, useState, useContext } from 'react';
 import { isNumber } from 'lodash-es';
 import { assign } from 'ndarray-ops';
 import { createMemo } from 'react-use';
 import { useFrame } from 'react-three-fiber';
-import type { HDF5Dataset, HDF5SimpleShape } from '../../providers/models';
+import { useAsyncResource } from 'use-async-resource';
+import type {
+  HDF5Dataset,
+  HDF5SimpleShape,
+  HDF5Id,
+  HDF5Value,
+} from '../../providers/models';
 import type { DimensionMapping } from '../../dataset-visualizer/models';
 import { getDomain } from './utils';
+import { ProviderContext } from '../../providers/context';
+
+export function useDatasetValue(id: HDF5Id): HDF5Value | undefined {
+  const { getValue } = useContext(ProviderContext);
+  const [valueReader] = useAsyncResource(getValue, id);
+  return valueReader();
+}
```

Comment:
The hook that does all the fetching. Much more concise than what used to be in `DatasetVisualizer`!

Reply:
I'll try to replace it with `use-suspender`, as discussed.

Reply:
Unlike `use-async-resource`'s `valueReader`, the hook generated with `use-suspender` calls the async function (i.e. `getValue`) every time, which suspends the rendering of the vis containers and leads to render flashes when slicing. We're better off staying with `use-async-resource`.